File: fix_spins.py
import os

files = [
    'frontend/src/pages/CryptoMarket.jsx',
    'frontend/src/pages/BondDetail.jsx',
    'frontend/src/pages/Dashboard.jsx',
]

for fpath in files:
    with open(fpath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Replace Spin tip pattern with nested div
    old_pattern = '<Spin size="large" tip="'
    if old_pattern in content:
        import re
        new_content = re.sub(
            r'<Spin size="large" tip="(.*?)" />',
            lambda m: f'<Spin size="large"><div>{m.group(1)}</div></Spin>',
            content
        )
        with open(fpath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f'Fixed {fpath}')
    else:
        print(f'Spin tip not found in {fpath}')

File: fix_spins.js
const fs = require('fs');
let c = fs.readFileSync('frontend/src/pages/CryptoMarket.jsx', 'utf8');

// Fix all Spin tip usages - replace <Spin size="large" tip="..." /> with nested div pattern
c = c.replace(
  /<Spin size="large" tip="(Loading chart\.\.\.)" \/>/g,
  '<Spin size="large"><div>$1</div></Spin>'
);

fs.writeFileSync('frontend/src/pages/CryptoMarket.jsx', c, 'utf8');
console.log('Fixed CryptoMarket.jsx Spin tips');

// Also fix BondDetail.jsx
let b = fs.readFileSync('frontend/src/pages/BondDetail.jsx', 'utf8');
b = b.replace(
  /<Spin size="large" tip="(Loading bond details\.\.\.)" \/>/g,
  '<Spin size="large"><div>$1</div></Spin>'
);
fs.writeFileSync('frontend/src/pages/BondDetail.jsx', b, 'utf8');
console.log('Fixed BondDetail.jsx Spin tips');

// Also fix Dashboard.jsx
let d = fs.readFileSync('frontend/src/pages/Dashboard.jsx', 'utf8');
d = d.replace(
  /<Spin size="large" tip="(Loading dashboard\.\.\.)" \/>/g,
  '<Spin size="large"><div>$1</div></Spin>'
);
fs.writeFileSync('frontend/src/pages/Dashboard.jsx', d, 'utf8');
console.log('Fixed Dashboard.jsx Spin tips');
